packages/nuxt/src/app/components/layout.ts

```diff
@@ -84,18 +46,53 @@ export default defineComponent({
 
       // We avoid rendering layout transition if there is no layout to render
       return _wrapIf(Transition, hasLayout && transitionProps, {
-        default: () => {
-          const layoutNode = _wrapIf(LayoutLoader, hasLayout && {
+        default: () => h(Suspense, { suspensible: true, onResolve: () => { nextTick(done) } }, {
+          default: () => _wrapIf(LayoutProvider, hasLayout && {
+            layoutProps: mergeProps(context.attrs, { ref: layoutRef }),
             key: layout.value,
             name: layout.value,
-            ...(process.dev ? { hasTransition: !!transitionProps } : {}),
-            ...context.attrs,
-            layoutRef
+            shouldProvide: !props.name,
+            hasTransition: !!transitionProps
           }, context.slots).default()
-
-          return layoutNode
-        }
+        })
       }).default()
     }
   }
 })
+
+const LayoutProvider = defineComponent({
+  name: 'NuxtLayoutProvider',
+  inheritAttrs: false,
+  props: ['name', 'layoutProps', 'hasTransition', 'shouldProvide'],
+  setup (props, context) {
+    // Prevent reactivity when the page will be rerendered in a different suspense fork
+    if (props.shouldProvide) {
+      // eslint-disable-next-line vue/no-setup-props-destructure
+      const name = props.name
+      provide('_layout', {
```

Comment:
I guess we could use a symbol as the injection key to avoid conflicts with user code?